probe/probe.go

```diff
@@ -0,0 +1,315 @@
+		if !strings.Contains(err.Error(), "not found") {
+			return err
+		}
+		fmt.Print("Karmor probe DaemonSet not found ...\n")
+	}
+
+
+	return nil
+}
+
+func PrintProbeResult(c *k8s.Client, o ProbeOptions) error{
+    if o.Full {
+        checkHostAuditSupport()
+        checkLsmSupport(getHostSupportedLSM())
+
+        if err := probeDaemonInstaller(c, o); err != nil {
+            return err
+        }
+        color.Yellow("\t Creating probe daemonset ...")
+        time.Sleep(2 * time.Second)
+        probeNode(c)
+        if err := probeDaemonUninstaller(c, o); err != nil {
+            return err
+        }
+	}else{
+        checkHostAuditSupport()
+        checkLsmSupport(getHostSupportedLSM())
+        color.Blue("To get full probe, a daemonset will be deployed in your cluster - This daemonset will be deleted after probing")
+        color.Blue("Use --full tag to get full probing")
+    }
+    return nil
+}
+
+
+func checkLsmSupport(supportedLSM string) {
+    fmt.Printf("\t Enforcement:") 
+    if strings.Contains(supportedLSM, "selinux"){
+        color.Yellow(" Partial (Supported LSMs: "+ supportedLSM + ") \n\t To have full enforcement support, apparmor must be supported")
+    }else if strings.Contains(supportedLSM, "apparmor"){
+        color.Green(" Full (Supported LSMs: "+ supportedLSM + ")")
+    }else{
+        color.Red(" None (Supported LSMs: "+ supportedLSM + ") \n\t To have full enforcement support, apparmor must be supported")
+    }
+}
+
+func getHostSupportedLSM() string {
+
+    b, err := os.ReadFile("/sys/kernel/security/lsm")
+    if err != nil {
+        log.Printf("an error occured when reading file")
+        return "none"
+    }
+    s := string(b)
+    return s
+}
+
+
+
+func kernelVersionSupported(kernelVersion string) bool {
+    if semver.Compare(kernelVersion, "4.14") >= 0 {
+        return true
+    }
+    return false;
+}
+
+func checkAuditSupport(kernelVersion string, kernelHeaderPresent bool) {
+
+    if kernelVersionSupported(kernelVersion) && kernelHeaderPresent{
+        color.Green(" Supported (Kernel Version " + kernelVersion )
+    }else if(kernelVersionSupported(kernelVersion)){
+        color.Red(" Not Supported : Kernel header must be present")
+    }else{
+        color.Red(" Not Supported (Kernel Version " + kernelVersion + " \n\t Kernel version must be greater than 4.14) and kernel header must be present")
+    }
+
+}
+
+func checkKernelHeaderPresent() bool{
+    //check if there's any directory /usr/src/$(uname -r)
+    var uname syscall.Utsname
+    if err := syscall.Uname(&uname); err == nil {
+
+        var path = ""
+        if _, err := os.Stat("/etc/redhat-release"); !os.IsNotExist(err) {
+            path = "/usr/src/"+int8ToStr(uname.Release[:])
+        }else{
+            path = "/usr/src/linux-headers-"+int8ToStr(uname.Release[:])
+        }
+       
+        if _, err := os.Stat(path); !os.IsNotExist(err) {
+            return true
+        }        
```

Comment:
I have updated the Kernel Header logic check here
https://github.com/kubearmor/KubeArmor/blob/main/KubeArmor/BPF/Makefile#L3-L11

Can you see if we can handle similarly here?
We meet need to add an additional mount point for "/lib/modules"